src/op/data.rs

```diff
@@ -142,6 +140,75 @@ pub fn missing(data: &Value, args: &Vec<&Value>) -> Result<Value, Error> {
     Ok(Value::Array(missing_keys))
 }
 
+/// Check whether a minimum threshold of keys are present in the data
+///
+/// Note that I think this function is confusingly named. `contains_at_least`
+/// might be better, or something like that. Regardless, it checks to see how
+/// many of the specified keys are present in the data. If there are equal
+/// to or more than the threshold value _present_ in the data, an empty
+/// array is returned. Otherwise, an array containing all missing keys
+/// is returned.
+pub fn missing_some(data: &Value, args: &Vec<&Value>) -> Result<Value, Error> {
+    let (threshold_arg, keys_arg) = (args[0], args[1]);
+
+    let threshold = match threshold_arg {
+        Value::Number(n) => n.as_u64(),
+        _ => None,
+    }
+    .ok_or(Error::InvalidArgument {
+        value: threshold_arg.clone(),
+        operation: "missing_some".into(),
+        reason: "missing_some threshold must be a valid, positive integer".into(),
+    })?;
+
+    let keys = match keys_arg {
+        Value::Array(keys) => Ok(keys),
+        _ => Err(Error::InvalidArgument {
+            value: keys_arg.clone(),
+            operation: "missig_some".into(),
+            reason: "missing_some keys must be an array".into(),
+        }),
+    }?;
+
+    let mut missing_keys: Vec<Value> = Vec::new();
+    let present_count = keys.into_iter().fold(Ok(0 as u64), |last, key| {
```

Comment:
Is there a reason we're hanlding `Option<u64>` in `present_count` I don't see any none scenarios?

Reply:
We're not. `present_count` is a `Result<u64, Error>` prior to the `?` on line 201. We're tracking the accumulator as a result because the fold function has potential error states that we'd like to bubble up. So since the first thing we do in the fold is `last?`, if the previous iteration was an error, we'll just continue propagating that error up to the eventual result of the fold (short-circuiting essentially).

Reply:
note that in other situations you might be able to do the same thing with a map to a result and then a `.collect::<Result<Vec<T>, E>>()`, since collect short-circuits on an iterator if it finds an error, _but_ in this case we need to pass along the accumulated present_count value, thus the `fold`. Also I'm still often bad about recognizing places where I could use the former instead of the latter.